### PR TITLE
Remove unused code from the fips module

### PR DESCRIPTION
**Pull request by slontis**
Now that we have checksums, it is a bit easier to spot source files that should not be in the fips module.

NOTE that the MD5 ASM sources are still in the checksum file for some reason.

This is a follow-on from PR #15164

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/s3_cbc.c
Comment (paulidale) [inline ssl/s3_cbc.c:206]: Why doesn't this generate a defined but not used error when building FIPS?

Both `u32toLE` and `tls1_md5_final_raw()` ought to be conditioned on FIPS.

File: crypto/md5/build.info
Comment (paulidale) [inline crypto/md5/build.info:35]: Might this and the following lines be the reason for the asm getting included?

**Review by paulidale — COMMENTED** [review]
Looks good.

File: crypto/md5/build.info
Comment (levitte) [reply crypto/md5/build.info:35]: No.  There's a more generic inclusion of perlasm files in `Configuration/unix-Makefile.tmpl`, for the very simple reason that we're not trying to do an exhaustive reconfiguration for every darn platform.  We *could* have a hard coded list instead, but that comes with a maintenance cost.

File: ssl/build.info
Comment (t8m) [inline ssl/build.info:43]: This needs to be based on #15171 instead. But that can wait until 15171 is merged as that will trigger a need to rebase anyway.

**Comment by levitte** [comment]
> NOTE that the MD5 ASM sources are still in the checksum file for some reason.

``` diff
diff --git a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
index d98c42c85e..c2a0de3a97 100644
--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -1167,7 +1167,6 @@ $(SRCDIR)/providers/fips.module.sources: \
          for x in crypto/bn/asm/*.pl crypto/bn/asm/*.S \
                   crypto/aes/asm/*.pl crypto/aes/asm/*.S \
                   crypto/ec/asm/*.pl \
-                  crypto/md5/asm/*.pl \
                   crypto/modes/asm/*.pl \
                   crypto/sha/asm/*.pl; do \
            echo "$$x"; \
```

File: ssl/s3_cbc.c
Comment (slontis) [reply ssl/s3_cbc.c:206]: tls1_md5_final_raw() already was.. u32toLE is a macro.. I have moved it inside the #ifndef

**Comment by slontis** [comment]
> CIs won't run with the conflict :(

Needs #15171 to merged anyway.. So may as well wait till then.

**Comment by t8m** [comment]
#15171 is merged now

**Review by levitte — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master after fixing the checksums.

**Pull request closed without merging**
